### ISSUE TITLE
Compiler FE: Enable CircleGRU Op

### ISSUE THREAD

**Issue by BalyshevArtem** [Samsung/ONE#12320]
Lets' enable Circle_GRU Op

- [ ] circle-schema
- [ ] res/CircleRecipes
- [ ] circledump (nothing to do)
- [ ] luci.lang
- [ ] luci.import
- [ ] luci.export
- [ ] luci.logex
- [ ] luci.service
- [ ] luci.partition
- [ ] luci.tests
- [ ] luci_interpreter - ?
- [ ] quantizer (verifier, fake quantization, const input) - ?
- [ ] luci.pass to fuse pattern into op
- [ ] circle2circle
- [ ] circle2circle-dredd-recipe-test
- [ ] luci-pass-value-test
- [ ] one-cmds

Related issue: #12263

**Comment by seanshpark** [comment]
from https://github.com/Samsung/ONE/pull/12319/files#r1452767255, 
with 6ee3e3ea32d3c69c5b27191135e72515978dba4a


About the name
- in draft, circle chef has `CircleOpCircleGRU` having two `Circle`
- IR is `CircleGRU`

in schema
- Op is `CIR_GRU`
- option is `CircleGRUOptions`

already existing ones
- we have circle only `CircleInstanceNorm` which can conflict when tflite introduces same name
- so does `GRU`

---

what I think
- current draft seems lack of consistency
- go with base name of `CirGru`
- chef: go with `CircleOpCircGru`
- schema: go with OP `CIR_GRU`, options `CirGruOptions`
- IR: go with `CircleCirGru`

@mhs4670go , @jinevening , @hseok-oh , any comments?

**Comment by jinevening** [comment]
Looks good to me.

> IR: go with CircleCirGru

It would be nice to add comments why "Cir" is repeated near the `CircleCirGru` class definition.

> circle2circle option fuse_gru looks OK for users

👍

**Comment by seanshpark** [comment]
> why "Cir" is repeated near the CircleCirGru class definition.

for me... I think start point would be the OP name in schema.
1/ `GRU` -> this is fine but may conflict when tflite introduces `GRU`
2/ `CIRCLE_GRU` -> looks OK but we may end up in IR like `CircleCircleGru` which gives two `Circle`
3/ `CIR_GRU` -> to prevent case 1/ and 2/, result is like this, which gives `CircleCirGru` IR

Any other proposals are welcome :)

**Comment by hseok-oh** [comment]
> from https://github.com/Samsung/ONE/pull/12319/files#r1452767255, with [6ee3e3e](https://github.com/Samsung/ONE/commit/6ee3e3ea32d3c69c5b27191135e72515978dba4a)
> 
> About the name
> 
>     * in draft, circle chef has `CircleOpCircleGRU` having two `Circle`
> 
>     * IR is `CircleGRU`
> 
> 
> in schema
> 
>     * Op is `CIR_GRU`
> 
>     * option is `CircleGRUOptions`
> 
> 
> already existing ones
> 
>     * we have circle only `CircleInstanceNorm` which can conflict when tflite introduces same name
> 
>     * so does `GRU`
> 
> 
> what I think
> 
>     * current draft seems lack of consistency
> 
>     * go with base name of `CirGru`
> 
>     * chef: go with `CircleOpCircGru`
> 
>     * schema: go with OP `CIR_GRU`, options `CirGruOptions`
> 
>     * IR: go with `CircleCirGru`
> 
>     * Fuse class: `FuseGRU` -> `FuseCirGru`
>       
>       * `circle2circle` option `fuse_gru` looks OK for users
> 
> 
> @mhs4670go , @jinevening , @hseok-oh , any comments?

I think we can use `GRU` (and `GRUOption`) in schema for this. If `TFLite` introduce `GRU`, TFLite's OP may be mapped to `GRU_TFLITE` (`GRUTFLiteOption`), `GRU_V2` (`GRUV2Option`) or other name in circle schema.

**Comment by mhs4670go** [comment]
I agree on using just `GRU` for the schema but if we want to make it easy to follow tflite schema update, using other naming looks okay. Maybe `CUSTOM_GRU` or something?

**Comment by BalyshevArtem** [comment]
> * go with base name of `CirGru`

Looks good to me 

> It would be nice to add comments why "Cir" is repeated near the `CircleCirGru` class definition.

👍
